internal/controllers/imagerepository_controller.go

```diff
@@ -516,8 +519,44 @@ func (r *ImageRepositoryReconciler) scan(ctx context.Context, obj *imagev1.Image
 		return 0, err
 	}
 
+	storedTags := make([]database.Tag, 0, len(filteredTags))
+
+	resCh := make(chan database.Tag, len(filteredTags))
+	var wg sync.WaitGroup
+	for _, tag := range filteredTags {
```

Comment:
What happens if there are 1000+ tags? Do we DDOS the registry with 1K HTTP requests all in parallel?

Reply:
Yeah, I went with a pretty simplistic approach towards it and had the same thought. We should use a goroutine pool.